config/initializers/devise.rb

```diff
@@ -258,8 +258,12 @@
   # Add a new OmniAuth provider. Check the wiki for more information on setting
   # up on your models and hooks.
   # config.omniauth :github, 'APP_ID', 'APP_SECRET', scope: 'user,public_repo'
-  config.omniauth :facebook, '241163829759918', '23c20a5525d87e610402a241386b537e', callback_url: 'http://localhost:3000/users/auth/facebook/callback'
-
+  if Rails.env == 'development' || Rails.env == 'test'
+    config.omniauth :facebook, '241163829759918', '23c20a5525d87e610402a241386b537e', callback_url: 'http://localhost:3000/users/auth/facebook/callback'
```

Comment:
Move the client id, secret and callback url to a environment variable. That way you don't need to check for the environment and, the most important, you are not uploading sensible data to the repo.